test/services/smart-action-hook.test.js

```diff
@@ -73,21 +62,164 @@ describe('services > smart-action-hook', () => {
           changedField: null,
         },
       );
-      expect(isSameDataStructure).toHaveBeenNthCalledWith(
-        1,
-        [{ ...field, value: null }],
-        [expected],
-        1,
-      );
+    });
+
+    describe('when user change fields array inside the hook', () => {
+      describe('when user add a field', () => {
+        it('should have one more field', async () => {
+          expect.assertions(2);
+
+          const { smartActionHook } = initContext().inject();
+          const field = {
+            field: 'myField',
+            type: 'String',
+            value: 'foo',
+          };
+          const addedField = {
+            field: 'myNewField',
+            type: 'String',
+            value: 'bar',
+          };
+
+          const fields = [field];
+          const expected = [field, addedField];
+
+          const hook = jest.fn(() => (expected));
+
+          const response = await smartActionHook.getResponse({}, hook, fields, {});
+          await expect(response).toStrictEqual(expected);
+          expect(hook).toHaveBeenNthCalledWith(
+            1,
+            {
+              fields,
+              record: {},
+              changedField: null,
+            },
+          );
+        });
+
+        it('should assign null value if no one is provided', async () => {
```

Comment:
```suggestion
        it('should assign null value if none is provided', async () => {
```

Reply:
All these are tests cases for incorrect usage, maybe a big `describe` would ease the readability